src/routes/social.js

```diff
@@ -0,0 +1,297 @@
+import express from 'express';
+import Course_reports from '../models/Course_reports';
+import Post_reports from '../models/Post_reports';
+import Courses from '../models/Courses';
+import { sendWebhookMessage } from '../utils/webhook_client';
+import { checkJwt } from '../auth';
+import { v4 as uuidv4 } from 'uuid';
+
+// route: "/api/v1/social"
+const router = express.Router();
+
+const get_self_vote_status = (post, user_id) => {
+  if(post.upvotes.includes(user_id)){
+    return 1;
+  }else if(post.downvotes.includes(user_id)){
+    return -1;
+  }else{
+    return 0;
+  }
+}
+
+
+router.get('/posts/:id/', checkJwt, async (req, res) => {
+  // get course social posts by course id
```

Comment:
by post id, not by course id?

Reply:
This method is created for future-proof. There is a GET post by course_id method below! 👌

Reply:
Ahh... I see.
You're talking about the comment right? I'll fix it...🥵

Reply:
Modified comment.